its/src/test/java/com/sonar/it/scanner/msbuild/RuntimeConfigTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.scanner.msbuild;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class RuntimeConfigTest {
+
+  @Test
+  public void TestRuntimConfigRollForward() throws IOException {
```

Comment:
This test never run on the build machine as it's not part to the test suite configured to run at build time. 

I'm also not sure if it checks the right thing since the `build` folder is created by the `ci-build` script that is supposed to run on dev machines.

Reply:
I've added https://github.com/SonarSource/sonar-scanner-msbuild/issues/1477 to handle this.

Reply:
good catch

is it because the S4NET codebase is missing a "ensure all tests are executed" test like https://github.com/SonarSource/sonar-dotnet/blob/master/its/src/test/java/com/sonar/it/shared/EnsureAllTestsRunBase.java

or is it more about the structure of the CI?

Reply:
I think it's about both. In this repository, we configure in the pom file which suite should run: https://github.com/SonarSource/sonar-scanner-msbuild/blob/master/its/pom.xml#L96

If there are tests missing in the test suite we will never run them and this test was missing.

We should, at some point, do an effort and create custom test runners for all our projects.